File: leetcode/top-interview-150/217/Solution.java
import java.util.HashMap;
import java.util.Map;

/**
 * CONTAINS DUPLICATE
 * Given an array num, return true if any value appears at least twice in the
 * array
 * and return false if every element is distinct.
 */

public class Solution {

    public static boolean containsDuplicate(int[] nums) {

        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
            if (map.get(num) > 1) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        int[] nums = { 1, 2, 3, 1 }; // false
        System.out.println(containsDuplicate(nums));
    }
}

File: leetcode/top-interview-150/217/Solution.go
package main

import "fmt"

func main() {
	nums := []int{1, 2, 3, 4, 5, 6}
	fmt.Print(containsDuplicate(nums))
}

func containsDuplicate(nums []int) bool {
	set := make(map[int]struct{})
	for _, num := range nums {
		if _, exists := set[num]; exists {
			return true
		}
		set[num] = struct{}{}
	}
	return false
}
